### PR TITLE
Allowing constructing CredentialsFile from string

### PR DESCRIPTION
**Pull request by benjaminch**
PR aiming to tackle issue described here: https://github.com/yoshidan/google-cloud-rust/issues/200

### CODE REVIEW

**Comment by yoshidan** [comment]
Please run 
```
cargo clippy --fix
cargo fmt
```
to fix CI.

**Comment by benjaminch** [comment]
Oupsie ! Fixing it now :)

**Comment by benjaminch** [comment]
done !

**Comment by yoshidan** [comment]
Overriding environment variables cause other tests to fail.
In the scope of your addition, only `test_credentials_file_new_from_str` is sufficient as a test, so could you please remove the other tests, leaving only `test_credentials_file_new_from_str` for the time being? I would appreciate it if you could delete the other tests.

**Comment by benjaminch** [comment]
Hey @yoshidan ! 
Would it be ok to use [a dedicated env var crates](https://crates.io/crates/temp-env) tackling those challenges? 
It will allow to set env vars, test it without any issue having tests running in parallel?

It allows to keep tests on this part :)

Cheers

File: foundation/auth/Cargo.toml [hunk 1]
```diff
@@ -36,7 +36,9 @@ hex = { version = "0.4", optional = true }
 tokio = { version = "1.32", features = ["test-util", "rt-multi-thread", "macros"]}
 tracing-subscriber = {version="0.3", features=["env-filter","std"]}
 ctor = "0.1"
-serial_test = "0.9"
+tempdir = "0.3.7"
```
Comment (yoshidan) [inline foundation/auth/Cargo.toml:39]: Cloud you remove tempdir?
tempdir is archived and has volunerability.
https://rustsec.org/advisories/RUSTSEC-2018-0017.html

Comment (benjaminch) [reply foundation/auth/Cargo.toml]: Ah yes, good catch, thanks :)

**Comment by yoshidan** [comment]
Thanks!

**Comment by benjaminch** [comment]
Hey @yoshidan ! Would you mind creating a new release with this change? 
Thanks a lot :)

**Comment by yoshidan** [comment]
Hi @benjaminch 
I published new release now.

**Pull request merged**